### PR TITLE
Device audit log

### PR DESCRIPTION
**Pull request by hardillb**
## Description

Add a Device specific Audit Log page and log entries

MVP covering logging of:

- credentials regenerated
- developer mode toggle
- editor access toggle
- device assigned to application/instance

Screenshot from 2024-01-05 13-10-33

## Related Issue(s)

#3268 

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Review by Steve-Mcl — CHANGES_REQUESTED** [review]
All loggers have matching tests, these should too (sowwy)

See `test/unit/forge/auditLog/application_spec.js` for example.

File: frontend/src/components/audit-log/AuditEntryIcon.vue [hunk 1]
```diff
@@ -198,7 +199,11 @@ const iconMap = {
         'team.device.developer-mode.enabled',
         'team.device.developer-mode.disabled',
         'team.device.remote-access.enabled',
-        'team.device.remote-access.disabled'
+        'team.device.remote-access.disabled',
+        'device.developer-mode.enabled',
+        'device.developer-mode.disabled',
+        'device.remote-access.enabled',
+        'device.remote-access.disabled'
```
[the remark this answers is not in the document]
Comment (hardillb) [reply frontend/src/components/audit-log/AuditEntryIcon.vue:206]: This is just making use of the same icon for the same events across all audit logs, to make recognition easier. This is correct

[the remark this answers is not in the document]
Comment (Steve-Mcl) [reply frontend/src/components/audit-log/AuditEntryIcon.vue:206]: > This is just making use of the same icon for the same events across all audit log

Devices have a device icon:

image


Whereas this PR makes all devices have the flask icon 
image

[the remark this answers is not in the document]
Comment (Steve-Mcl) [reply frontend/src/components/audit-log/AuditEntryIcon.vue:206]: I suppose it kinda makes sense as these are developer mode interactions, but looks odd to me.

@joepavitt any opinion here?

[the remark this answers is not in the document]
Comment (Steve-Mcl) [reply frontend/src/components/audit-log/AuditEntryIcon.vue:206]: Ah, my bad, I though ALL new entries were under beaker, but they are not (just spotted `'device.credentials.generated'` is under the `device` icon.

**Review by Steve-Mcl — APPROVED** [review]

**Pull request merged**